### PR TITLE
Extend API for NFT1

### PR DESCRIPTION
**Pull request by ytx1991**

### CODE REVIEW

**Comment by trepca** [comment]
looks like there are some linting and python 3.7 support issues, can you merge my latest changes

File: chia/wallet/nft_wallet/nft_info.py [hunk 1]
```diff
@@ -59,6 +62,9 @@ class NFTInfo(Streamable):
     chain_info: str
     """Information saved on the chain in hex"""
 
+    mint_height: uint32
```
[the remark this answers is not in the document]
Comment (ytx1991) [reply chia/wallet/nft_wallet/nft_info.py:65]: Discussed in the meeting

File: chia/wallet/nft_wallet/nft_info.py [hunk 2]
```diff
@@ -59,6 +62,12 @@ class NFTInfo(Streamable):
...
+    """Block height of the NFT minting"""
+
+    support_did: bool
```
Comment (trepca) [inline chia/wallet/nft_wallet/nft_info.py:68]: this should probably be `supports_did`?

File: chia/rpc/wallet_rpc_api.py [hunk 3]
```diff
@@ -1363,7 +1365,14 @@ async def nft_mint_nft(self, request) -> Dict:
...
         )
         fee = uint64(request.get("fee", 0))
-        spend_bundle = await nft_wallet.generate_new_nft(metadata, royalty_puzhash, target_puzhash, fee=fee)
+        spend_bundle = await nft_wallet.generate_new_nft(
+            metadata,
+            royalty_puzhash,
+            target_puzhash,
+            uint16(request.get("royalty_percentage", 0)),
+            request.get("use_did", True),
+            fee,
```
Comment (paninaro) [inline chia/rpc/wallet_rpc_api.py:1374]: If the minting wallet doesn't have a DID associated with it, should it be possible to set the royalty info?

Also, should `request.get("use_did", True),` be something like `request.get("use_did", wallet.get_did() != None)` to handle the same case where the NFT wallet doesn't have a DID?

**Review by paninaro — DISMISSED** [review]
lgtm pending Seb's review

File: chia/rpc/wallet_rpc_api.py (hunk 3 above)
Comment (ytx1991) [reply chia/rpc/wallet_rpc_api.py:1374]: It doesn't matter here. If the DID is none, the use_did will be ignored.

**Review by paninaro — APPROVED** [review]
lgtm

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**